tests/Microsoft.Identity.Test.Unit/PublicApiTests/TenantIdTests.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Test.Common;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Unit.PublicApiTests
+{
+    [TestClass]
+    public class TenantIdTests
+    {
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            TestCommon.ResetInternalStaticCaches();
+        }
+
+        [DataTestMethod]
+        [DataRow(TestConstants.AuthorityCommonTenant, "common", DisplayName = "Common endpoint")]
+        [DataRow(TestConstants.AuthorityNotKnownCommon, "common", DisplayName = "Common endpoint")]
+        [DataRow(TestConstants.AuthorityHomeTenant, "home", DisplayName = "Home endpoint")]
+        [DataRow(TestConstants.AuthorityGuestTenant, "guest", DisplayName = "Guest endpoint")]
+        [DataRow(TestConstants.AuthorityOrganizationsTenant, "organizations", DisplayName = "Organizations endpoint")]
+        [DataRow(TestConstants.AadAuthorityWithTestTenantId, TestConstants.AadTenantId, DisplayName = "Tenant Id")]
+        [DataRow(TestConstants.AuthorityConsumersTenant, "consumers", DisplayName = "Consumers Endpoint")]
+        [DataRow(TestConstants.AuthorityConsumerTidTenant, TestConstants.MsaTenantId, DisplayName = "Tenant Id")]
+        [DataRow(TestConstants.AuthorityNotKnownTenanted, TestConstants.Utid, DisplayName = "Tenant Id")]
+        [DataRow(TestConstants.AuthorityUtidTenant, TestConstants.Utid, DisplayName = "Tenant Id")]
+        [DataRow(TestConstants.AuthorityUtid2Tenant, TestConstants.Utid2, DisplayName = "Tenant Id")]
```

Comment:
Can you please add a few more data rows in there: 

- B2C (B2CAuthority B2CLoginAuthority B2CLoginAuthorityUsGov B2CCustomDomain)
- DSTS (this is an internal type of authority). It behaves like AAD. https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/tests/Microsoft.Identity.Test.Unit/CoreTests/InstanceTests/DstsAuthorityTests.cs#L22
- A sovereign cloud like `https://login.partner.microsoftonline.cn/tenantid`

Reply:
Would you mind checking again?